src/python/WMCore/Credential/Proxy.py

```diff
@@ -159,6 +159,8 @@ def setEnv(self, cmd):
         if self.uisource is not None and len(self.uisource) > 0:
             ret = 'source ' + self.uisource + ' && ' + ret
         if self.cleanEnvironment:
+            # Need to escape ' in cmd
+            ret = ret.replace("'", r"'\''")
```

Comment:
Hi Justas, 

I meant whether there need additional sing quote?
``` python
ret = ret.replace("'", r"'\''") # this produces "Tom'\\''s house"
ret = ret.replace("'", r"\'") # this produces "Tom\\'s house"
```
I was just wondering the first one is what you want (I thought it should be the second case)
first one produces